render/gles2/renderer.c

```diff
@@ -211,20 +211,9 @@ static bool wlr_gles2_buffer_is_drm(struct wlr_renderer *_renderer,
 		EGL_TEXTURE_FORMAT, &format);
 }
 
-static void rgba_to_argb(uint32_t *data, size_t height, size_t stride) {
-	size_t n = height*stride/4;
-	for (size_t i = 0; i < n; ++i) {
-		uint32_t v = data[i];
-		uint32_t rgb = (v & 0xffffff00) >> 8;
-		uint32_t a = v & 0x000000ff;
-		data[i] = rgb | (a << 24);
-	}
-}
-
 static void wlr_gles2_read_pixels(struct wlr_renderer *renderer, int x, int y,
 		int width, int height, void *out_data) {
-	glReadPixels(x, y, width, height, GL_RGBA, GL_UNSIGNED_BYTE, out_data);
-	rgba_to_argb(out_data, height, width*4);
+	glReadPixels(x, y, width, height, GL_BGRA_EXT, GL_UNSIGNED_BYTE, out_data);
```

Comment:
BGRA != ARGB, is this change intended?

Reply:
Yes. OpenGL will actually read the pixels in the reverse order than you expect.
It's what happens in the DRM cursor function as well.

Reply:
Not working for me, here's what the screenshooter outputs:

![wayland-screenshot](https://user-images.githubusercontent.com/506932/31856141-cc1258dc-b6ba-11e7-8b92-ecf4e1336221.png)

Reply:
Though I know that OpenGL reads images upside down, so data must be y-flipped (that happens in `wlr_screenshooter`).

Reply:
I rolled back that commit